homeassistant/components/climate/tado.py

```diff
@@ -75,8 +78,12 @@ def create_climate_device(tado, hass, zone, name, zone_id):
 
     if ac_mode:
         temperatures = capabilities['HEAT']['temperatures']
-    else:
+    elif 'temperatures' in capabilities:
+        temperatures = capabilities['temperatures']
         temperatures = capabilities['temperatures']
+    else:
+        _LOGGER.debug("Recieved zone %s has no temperature; not adding", name)
+        return False
```

Comment:
It's a bad design to return False if you want have a object. Do only `return` and check if device is valid above